crates/biome_formatter/src/lib.rs

```diff
@@ -727,6 +771,9 @@ pub trait FormatOptions {
     /// Whether to insert spaces around brackets in object literals. Defaults to true.
     fn bracket_spacing(&self) -> BracketSpacing;
 
+    /// Whether to enforce collapsing object literals when possible. Defaults to preserve.
+    fn object_wrap(&self) -> ObjectWrap;
```

Comment:
Should `ObjectWrap` be part of `FormatOptions`? The responsibility of `biome_formatter` is to convert IR into a string, it cannot understand what `object_wrap` is

Reply:
@ah-yu
Biome (and Prettier) formatter has a heustric that keeps expanding a object-like expressions when it already has a leading newline.

For example, **both** code below are considered as formatted:

```js
const foo = { bar: "baz" };
```

```js
const foo = {
  bar: "baz",
};
```

`ObjectWrap` option removes this heustric to improve code consistency. In this case, only the code below are considered as formatted:

```js
const foo = { bar: "baz" };
```

(Note that it will be the expanded style if the object is longer than the max line width.)


It was recently added in [Prettier 3.5](https://prettier.io/blog/2025/02/09/3.5.0) and I ported it in #5068. I just copied the name of Prettier option, so any suggestion for a new name is welcome :)

Reply:
Ah, I now understand what you meant after looking at #5088. It makes sense for me, I will do that for objectWrap once the PR got merged.